spec.html

```diff
@@ -9337,6 +9337,24 @@ <h1>Static Semantics: UTF16Decode ( _lead_, _trail_ )</h1>
         1. Return the code point _cp_.
       </emu-alg>
     </emu-clause>
+
+    <emu-clause id="sec-codepointat" aoid="CodePointAt">
+      <h1>Static Semantics: CodePointAt ( _string_, _position_ )</h1>
+      <p>The abstract operation CodePointAt interprets a String _string_ as a sequence of UTF-16 encoded code points, as described in <emu-xref href="#sec-ecmascript-language-types-string-type"></emu-xref>, and reads from it a single code point starting with the code unit at index _position_. When called, the following steps are performed:</p>
+      <emu-alg>
+        1. Let _size_ be the length of _string_.
+        1. Assert: _position_ &ge; 0 and _position_ &lt; _size_.
+        1. Let _first_ be the code unit at index _position_ within _string_.
+        1. Let _cp_ be the code point whose numeric value is that of _first_.
+        1. If _first_ is not a <emu-xref href="#leading-surrogate"></emu-xref> or <emu-xref href="#trailing-surrogate"></emu-xref>, return a new Record { [[CodePoint]]: _cp_, [[CodeUnitCount]]: 1, [[IsUnpairedSurrogate]]: *false* }.
+        1. If _first_ is a <emu-xref href="#leading-surrogate"></emu-xref> and _position_ + 1 &lt; _size_, then
+          1. Let _second_ be the code unit at index _position_ + 1 within _string_.
+          1. If _second_ is a <emu-xref href="#trailing-surrogate"></emu-xref>, then
+            1. Let _cp_ be ! UTF16Decode(_first_, _second_).
```

Comment:
The previous `_cp_` is still "in scope", so this step should either use "Set" or a different alias.

Reply:
Resolved by switching back to a linear approach.